### PR TITLE
24/9/19 change: return control after face-down

### PR DESCRIPTION
**Pull request by salix5**
https://db.ygoresources.com/qa#9423
https://db.ygoresources.com/qa#11049
https://db.ygoresources.com/qa#16310
https://db.ygoresources.com/qa#24100

Before:
If the control is changed by an equip card, the control will not return after flipped face-down.

Now:
The control will return.

@mercury233 
@purerosefallen

### CODE REVIEW

File: card.cpp [hunk 1]
```diff
@@ -2165,19 +2165,7 @@ void card::reset(uint32 id, uint32 reset_type) {
...
-				new_effect->owner = this;
-				new_effect->handler = this;
-				new_effect->type = EFFECT_TYPE_SINGLE;
-				new_effect->code = EFFECT_SET_CONTROL;
-				new_effect->value = current.controler;
-				new_effect->flag[0] = EFFECT_FLAG_CANNOT_DISABLE;
-				new_effect->reset_flag = RESET_EVENT | 0xec0000;
-				this->add_effect(new_effect);
-			}
+			refresh_control_status();
```
Comment (mercury233) [inline card.cpp:2168]: is this necessary?

Comment (salix5) [reply card.cpp]: Thank you.

File: libcard.cpp [hunk 2]
```diff
@@ -3125,7 +3125,7 @@ int32 scriptlib::card_is_immune_to_effect(lua_State *L) {
 	check_param(L, PARAM_TYPE_EFFECT, 2);
 	card* pcard = *(card**) lua_touserdata(L, 1);
 	effect* peffect = *(effect**) lua_touserdata(L, 2);
-	lua_pushboolean(L, !pcard->is_affect_by_effect(peffect));
+	lua_pushboolean(L, (int)!pcard->is_affect_by_effect(peffect));
```
Comment (mercury233) [inline libcard.cpp:3128]: better move to another pull request

**Review by mercury233 — APPROVED** [review]
